WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java

```diff
@@ -45,16 +46,47 @@ public static void migrateAppWideMobileEditorPreferenceToRemote(final AccountSto
             return;
         }
 
-        // In a later version we might override mobile_editor setting if it's set to `aztec` and show a specific notice
-        // for these users ("We made a lot of progress on the block editor and we think it's now better than
-        // the classic editor, we switched it on, but you can change the configuration in your Site Settings").
-        // ^ This code should be here.
-
         // If the user is already in the rollout group, we can skip this the migration.
         if (AppPrefs.isUserInGutenbergRolloutGroup()) {
             return;
         }
 
+        // -------- Rollout: Phase 2 -------
+        //
+        // If user has any Aztec enabled sites, we'll migrate their sites to Gutenberg and schedule showing popup.
+        if (atLeastOneSiteHasAztecEnabled(siteStore)) {
+            // Randomly pick the user in the rollout group
+            if (accountStore.getAccount().getUserId() % 100 >= (100 - GB_ROLLOUT_PERCENTAGE_PHASE_2)) {
+                if (!NetworkUtils.isNetworkAvailable(WordPress.getContext())) {
+                    // If the network is not available, abort. We can't update the remote setting.
+                    return;
+                }
+
+                for (SiteModel site : siteStore.getSites()) {
+                    // Show "phase 2" dialog on sites that get switched from aztec to gutenberg
+                    if (TextUtils.equals(site.getMobileEditor(), AZTEC_EDITOR_NAME)) {
+                        AppPrefs.setShowGutenbergInfoPopupPhase2ForNewPosts(site.getUrl(), true);
+                        // Will show the popup again, even if it was displayed in the past
+                        AppPrefs.setGutenbergInfoPopupDisplayed(site.getUrl(), false);
+                    }
+
+                    // Show "phase 1" dialog on sites that get switched from "empty" (no pref) to gutenberg
+                    if (TextUtils.isEmpty(site.getMobileEditor())) {
```

Comment:
What do you think about adding a `continue` statement to finish the loop iteration in the earlier conditional, or alternatively, merge the conditionals with an `else if`?

Reply:
Yes they're mutually exclusive and a `else if` would work, but I preferred to keep it like this for readability. The `isEmpty` check won't have any performance impact.

Reply:
I'd recommend keeping them in separate blocks for readability :nod:, that's why I mentioned using a `continue` statement first. The end goal is to make the intention very clear, and probably even protect from future code modifications that could involuntarily change the execution flow. Not a blocker if you still prefer keeping the code as is though.

Reply:
Addressed with 4204171f2d2b1208e2f770c59621b8bf2c8cab9d.